src/python/pants/help/help_printer.py

```diff
@@ -35,7 +36,12 @@ def __init__(
     union_membership: UnionMembership,
   ) -> None:
     self._options = options
-    self._help_request = help_request or self._options.help_request
+    self._help_request = cast(HelpRequest, help_request or self._options.help_request)
+    if self._help_request is None:
+      raise AssertionError(
+        "HelpPrinter was instantiated without any corresponding `help_request`. This means that "
+        "there would be nothing to print and that there was a mistake from the call site."
+      )
```

Comment:
This was necessary to get MyPy happy. Technically, due to the `cast`, we don't actually need to check that `self._help_request` is `None`, but I figure defensive programming is a good thing here.